ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_initiator_group.py

```diff
@@ -0,0 +1,379 @@
+
+
+def delete_igroup(
+        client_obj,
+        initiator_group_name):
+
+    if utils.is_null_or_empty(initiator_group_name):
+        return (False, False, "Initiator group deletion failed as it is not present.", {})
+
+    try:
+        # see if the igroup is already present
+        ig_resp = client_obj.initiator_groups.get(id=None, name=initiator_group_name)
+        if ig_resp is not None:
+            client_obj.initiator_groups.delete(ig_resp.attrs.get("id"))
+            return (True, True, f"Successfully deleted initiator group '{initiator_group_name}'.", {})
+        elif ig_resp is None:
+            return (False, False, f"Initiator group '{initiator_group_name}' is not present on array.", {})
+        else:
+            return (False, False, f"Failed to delete initiator group '{initiator_group_name}'.", {})
+    except Exception as ex:
+        return (False, False, f"Initiator group deletion failed | {ex}", {})
+
+
+def main():
+
+    fields = {
+        "state": {
+            "required": True,
+            "choices": ['present',
+                        'absent',
+                        'create'
+                        # 'append'
+                        ],
+            "type": "str"
+        },
+        "name": {
+            "required": True,
+            "type": "str",
+            "no_log": False
+        },
+        "description": {
+            "required": False,
+            "type": "str",
+            "no_log": False
+        },
+        "access_protocol": {
+            "choices": ['iscsi',
+                        'fc'
+                        ],
+            "required": False,
+            "type": "str",
+            "no_log": False
+        },
+        "host_type": {
+            "required": False,
+            "type": "str",
+            "no_log": False
+        },
+        "fc_tdz_ports": {
+            "required": False,
+            "type": "list",
+            "no_log": False,
+        },
+        "target_subnets": {
+            "required": False,
+            "type": "list",
+            "no_log": False,
+        },
+        "iscsi_initiators": {
+            "required": False,
+            "type": "list",
+            "no_log": False,
+        },
+        "fc_initiators": {
+            "required": False,
+            "type": "list",
+            "no_log": False,
+            "default": None
+        },
+        "app_uuid": {
+            "required": False,
+            "type": "str",
+            "no_log": False
+        },
+        "metadata": {
+            "required": False,
+            "type": "dict",
+            "default": None
+        }
+    }
+
+    default_fields = utils.basic_auth_arg_fields()
+    fields.update(default_fields)
+    required_if = [('state', 'create', ['access_protocol'])]
+    module = AnsibleModule(argument_spec=fields, required_if=required_if)
+    if client is None:
+        module.fail_json(msg='the python nimble-sdk module is required.')
```

Comment:
Use correct grammar.

Reply:
have renamed to "Python nimble-sdk could not be found."
is that ok?